python_files/ranking_algos.py

```diff
@@ -0,0 +1,365 @@
+                predictors = [_ for _ in self.dataset.columns if _ not in response]
+                X = self.dataset[predictors]
+                y = self.dataset[response]
+                break
+
+            except KeyError:
+                print("Please enter in a valid column as your response")
+
+        return X, y
+
+    # determining column data type
+    def data_type(self, feat):
+        X_type = feat.convert_dtypes().dtypes
+
+        if X_type == np.float64:
+            X_type = "continuous"
+        elif type(X_type) == pd.Int64Dtype and np.array_equal(feat.unique(), [0, 1]):
+            X_type = "binary"
+        else:
+            X_type = "categorical"
+        print(f"Predictor: {feat.name} {X_type} data type")
+
+        return X_type
+
+    # linear regression model for a continuous response
+    def linear_regression_model(self, y, predictor):
+        lin_regr_model = statsmodels.api.OLS(y, predictor)
+        lin_regr_model_fitted = lin_regr_model.fit()
+        print(lin_regr_model_fitted.summary())
+
+        # Get the stats
+        t_val = round(lin_regr_model_fitted.tvalues[1], 6)
+        p_val = "{:.6e}".format(lin_regr_model_fitted.pvalues[1])
+
+        return t_val, p_val
+
+    # logisitic regression model for a boolean/categorical response
+    def logistic_regression_model(self, y, predictor):
+        log_regr_model = statsmodels.api.Logit(y, predictor)
+        log_regr_model_fitted = log_regr_model.fit()
+        print(log_regr_model_fitted.summary())
+
+        # Get the stats
+        t_val = round(log_regr_model_fitted.tvalues[1], 6)
+        p_val = "{:.6e}".format(log_regr_model_fitted.pvalues[1])
+
+        return t_val, p_val
+
+    # random forest model importance (classifier and regressor)
+    # @classmethod
+    def random_forest_model(self, X, y, y_type):
+        X_train, X_test, y_train, y_test = train_test_split(
+            X, np.ravel(y), test_size=0.2, random_state=0
+        )
+
+        if y_type == "binary":
+            rf_model = RandomForestClassifier(random_state=123)
+            rf_model_fitted = rf_model.fit(X_train, y_train)
+
+        else:
+            rf_model = RandomForestRegressor(n_estimators=100, random_state=123)
+            rf_model_fitted = rf_model.fit(X_train, y_train)
+
+        return rf_model_fitted
+
+    # continous response and continuous predictor plot
+    def plot_cont_resp_cont_pred(self, feat, y, y_name, **kwargs):
+        stat_text = f'(t-value={kwargs["t_val"]}) (p-value={kwargs["p_val"]})'
+        title_text = "Continuous Response by Continuous Predictor"
+        fig = px.scatter(x=feat, y=y, trendline="ols")
+        fig.update_layout(
+            title=f"{title_text}: {stat_text}",
+            xaxis_title=f"Predictor: {feat.name}",
+            yaxis_title=f"Response: {y_name}",
+        )
+        fig.show()
+        fig.write_html(
+            file=f"plots/{feat.name}_scatter_plot.html", include_plotlyjs="cdn"
```

Comment:
add ./ for relative path other wise i get an error when running??

Reply:
FileNotFoundError: [Errno 2] No such file or directory:

Reply:
or absolute path

Reply:
or import os

Reply:
ok fixed it thank you!